### PR TITLE
Shop Filters

### LINKED ISSUE #382: City shops filters
Not 100% necessary but a QoL thing as TT has added lots of filters everywhere.

Add filters to city shops where you could filter by
1) Stock: min-max (auto-populated)
2) Buy price: min-max (auto-populated)
3) Profit: min-max (auto-populated)

As I said, not anything critical, but it'd be much more effective when the negative profit items & items with stock less than 100 are automatically hidden as you load the page for example.

### PR DESCRIPTION
**Pull request by Sashank999**
Closes #382 .

### CODE REVIEW

File: extension/scripts/features/shop-filters/ttShopFilters.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	async function filtering() {
+		await requireElement(".item-desc");
+
+		const checkboxes = localFilters.filters.getSelections();
+		const hideLoss = checkboxes.includes("hideLoss");
+		const hideUnder100 = checkboxes.includes("hideUnder100");
+
+		await ttStorage.change({ filters: { shops: { hideLoss, hideUnder100 } } });
+
+		for (const li of document.findAll(".buy-items-wrap .items-list > li:not(.empty, .clear)")) {
```
Comment (DeKleineKobini) [inline extension/scripts/features/shop-filters/ttShopFilters.js:57]: What what, does that work? Didn't know that, nice one.

File: extension/scripts/global/functions/formatting.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 "use strict";
 
+const Regexes = {
```
Comment (DeKleineKobini) [inline extension/scripts/global/functions/formatting.js:3]: Should be called `REGEXES`, as it's a constant.

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]
Great already, only a very small change needed.

File: extension/scripts/features/shop-filters/ttShopFilters.js (hunk 1 above)
Comment (Sashank999) [reply extension/scripts/features/shop-filters/ttShopFilters.js:57]: Took some time to find it but it worked.

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**